src/Chapter2.hs

```diff
@@ -621,7 +633,13 @@ Write a function that takes elements of a list only on even positions.
 >>> takeEven [2, 1, 3, 5, 4]
 [2,3,4]
 -}
-takeEven = error "takeEven: Not implemented!"
+takeEven :: [Int] -> [Int]
+takeEven = go []
+  where
+    go :: [Int] -> [Int] -> [Int]
+    go acc [] = acc
+    go acc [x] = x : acc
+    go acc (x : _ : xs) = go (x : acc) xs
```

Comment:
By adding elements to `acc`, you get the list in reverse order, So you need to use the `reverse` function in the end. But there's a better solution actually. You can just add elements to the beginning of the result. Thus, you will get all elements in the proper order and don't need to use `reverse`. Morever. you can notice that the `acc` variable becomes redundant
```suggestion
takeEven = go []
  where
    go :: [Int] -> [Int] -> [Int]
    go acc [] = acc
    go acc [x] = x : acc
    go acc (x : _ : xs) = x : go acc xs
```